### PR TITLE
Clone Agent

### PR DESCRIPTION
**Pull request by Atralupus**
It can replace planet-clicker [Agent](https://github.com/planetarium/planet-clicker/blob/master/planet-clicker/Assets/LibplanetUnity/Agent.cs)
This class handles action, transaction, etc. in the background.

### CODE REVIEW

File: Libplanet.Unity/Utils.cs [hunk 1]
```diff
@@ -6,7 +6,7 @@
 using Libplanet.Store;
 using Libplanet.Unity;
 
-namespace LibplanetUnity
+namespace Libplanet.Unity
```
Comment (greymistcube) [inline Libplanet.Unity/Utils.cs:9]: 👍

File: Libplanet.Unity/MonoSingleton.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
+#nullable disable
+using UnityEngine;
+
+namespace Libplanet.Unity
+{
+    /// <summary>
+    /// MonoSingleton.
+    /// </summary>
+    /// <typeparam name="T">The type.</typeparam>
```
Comment (greymistcube) [inline Libplanet.Unity/MonoSingleton.cs:9]:
```suggestion
    /// <summary>
    /// A singleton implementation with <see cref="MonoBehaviour"/> as its base class.
    /// </summary>
    /// <typeparam name="T">The class to wrap with this singleton pattern.</typeparam>
```

I'd say single word summaries are going bit too far. 😐

Comment (Atralupus) [reply Libplanet.Unity/MonoSingleton.cs:9]: Oops, I'm sorry. I forgot to write the document
Thanks you 👍

File: Libplanet.Unity/MonoSingleton.cs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    public abstract class MonoSingleton<T> : MonoBehaviour
+        where T : MonoSingleton<T>
+    {
+        private static readonly object Lock = new object();
+        private static T _instance = null;
+        private static bool _applicationIsQuitting;
+
+        /// <summary>
+        /// MonoSingleton.
+        /// </summary>
```
Comment (greymistcube) [inline Libplanet.Unity/MonoSingleton.cs:19]:
```suggestion
        /// <summary>
        /// The singleton instance of <see cref="MonoSingleton{T}"/>.
        /// </summary>
```

Comment (Atralupus) [reply Libplanet.Unity/MonoSingleton.cs:19]: I'll documentaiton this file pls wait 🙏

**Review by greymistcube — APPROVED** [review]

**Pull request merged**